packages/graphql-api/src/resolvers/authenticate.ts

```diff
@@ -0,0 +1,26 @@
+import { AccountsServer } from "@accounts/server";
+import { IResolverContext } from "../types/graphql";
+
+export const serviceAuthenticate = (accountsServer: AccountsServer) => async (
+  _,
+  args: GQL.IAuthenticateOnMutationArguments,
+  ctx: IResolverContext
+) => {
+  const { serviceName, params } = args;
+
+  // ! No IP and userAgent, using fake ones!
+  // TODO: discussion needed
```

Comment:
I guess we can only take these params from the context no? ..

Reply:
@pradel I'm not sure if we should pass those params into the context, it's totally up to the developer to pass these arguments in right?

That's a different location (somewhere near the server / express route config) other than initializing the schema.

But we don't seem to have many options 😉

Reply:
@pradel Can we make `ip` and `userAgent` optional?

If yes, then user can optionally pass in related info. We can through out a warning for requiring `ip` and `userAgent` if needed.

Then we may let user to pass in a `IncomingMessage` instance (`req`) or pass in a portion of it, `ip` and `userAgent`.

Reply:
I don't know, there is already the `JSAccountsContext` function, can we extend it to take the user agent and ip and inject them in the context? Looks like the easiest solution to me.

Reply:
@pradel Resolved by using just `JSAccountsContext`.